.github/workflows/cover.yaml

```diff
@@ -0,0 +1,49 @@
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+    branches:
+    - main
+
+jobs:
+  coverage:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    - name: setup filesystem
+      run: |
+        go env
+        mkdir -p ${GITHUB_WORKSPACE}/bin
+        mkdir -p ${GITHUB_WORKSPACE}/tmp
+    - name: install gocov
+      run: |
+        export PATH=$PATH:${GITHUB_WORKSPACE}/bin
+        go env
+        mkdir -p ${GITHUB_WORKSPACE}/tmp/gocov
+        pushd ${GITHUB_WORKSPACE}/tmp/gocov
+        go mod init tmp
+        go get github.com/axw/gocov/...
+        cp $(go env GOPATH)/bin/gocov ${GITHUB_WORKSPACE}/bin/gocov
+        popd
+        rm -rf ${GITHUB_WORKSPACE}/tmp/gocov
+        file ${GITHUB_WORKSPACE}/bin/gocov
+    - name: install gocov-xml
+      run: |
+        export PATH=$PATH:${GITHUB_WORKSPACE}/bin
```

Comment:
maybe use something like https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/scripts/go_install.sh to dry this up.